src/Common/EnvironmentProxyConfigurationResolver.cpp

```diff
@@ -56,9 +68,22 @@ ProxyConfiguration EnvironmentProxyConfigurationResolver::resolve()
         host,
         ProxyConfiguration::protocolFromString(scheme),
         port,
-        useTunneling(request_protocol, ProxyConfiguration::protocolFromString(scheme), disable_tunneling_for_https_requests_over_http_proxy),
-        request_protocol
+        useTunneling(protocol, ProxyConfiguration::protocolFromString(scheme), disable_tunneling_for_https_requests_over_http_proxy),
```

Comment:
This is a problem actually. Since the creation of the configuration is now static, it happens only once. `disable_tunneling_for_https_requests_over_http_proxy` is a ClickHouse setting that could be changed in runtime, but it wouldn't be taken into account..